### PR TITLE
Feat import project

### LINKED ISSUE #16: Import should be implemented for all resource types
```
terraform import "env0_configuration_variable.test1" b94f954c-0b06-4fb4-8cb5-ed5161c99480
env0_configuration_variable.test1: Importing from ID "b94f954c-0b06-4fb4-8cb5-ed5161c99480"...

Error: Not implemented
```

:(

### PR DESCRIPTION
**Pull request by eranelbaz**
PR 3/4 for https://github.com/env0/terraform-provider-env0/issues/16
Enable `terraform import env0_project` using name or uuid or resource

### CODE REVIEW

**Comment by eranelbaz** [comment]
[TF Provider - Add import ability to all existing resources](https://trello.com/c/dn6qDaJ0/3287-tf-provider-add-import-ability-to-all-existing-resources)

File: env0/resource_project.go [hunk 1]
```diff
@@ -73,15 +74,19 @@ func resourceProjectDelete(ctx context.Context, d *schema.ResourceData, meta int
...
-	project, err := apiClient.Project(id)
-	if err != nil {
-		return nil, err
+	var getErr diag.Diagnostics
+	_, uuidErr := uuid.Parse(id)
+	if uuidErr == nil {
+		_, getErr = getProjectById(id, meta)
+	} else {
+		var project client.Project
+		project, getErr = getProjectByName(id, meta)
```
Comment (roni-frantchi) [inline env0/resource_project.go:84]: I'd add some logging here a-la: "parameter does not seem to be a valid env0 project id - trying to import by name xxx".

File: env0/data_project.go [hunk 2]
```diff
@@ -77,3 +67,32 @@ func dataProjectRead(ctx context.Context, d *schema.ResourceData, meta interface
...
+
+func getProjectByName(name interface{}, meta interface{}) (client.Project, diag.Diagnostics) {
+	apiClient := meta.(*client.ApiClient)
+	projects, err := apiClient.Projects()
+	var project client.Project
+
+	if err != nil {
+		return client.Project{}, diag.Errorf("Could not query project by name: %v", err)
+	}
+	for _, candidate := range projects {
```
Comment (roni-frantchi) [inline env0/data_project.go:79]: If we go by name, we should throw if there's more than one project matching the name

File: env0/resource_project.go (hunk 1 above)
Comment (roni-frantchi) [reply env0/resource_project.go]: https://github.com/env0/terraform-provider-env0/pull/42/commits/00a2a8b6117b316725800e470b04b4f63f690ce9

File: env0/data_project.go (hunk 2 above)
Comment (roni-frantchi) [reply env0/data_project.go:82]: https://github.com/env0/terraform-provider-env0/pull/42/commits/00a2a8b6117b316725800e470b04b4f63f690ce9

**Review by roni-frantchi — APPROVED** [review]

**Pull request merged**